lib/resty/core/phase.lua

```diff
@@ -0,0 +1,55 @@
+local ffi = require 'ffi'
+local C = ffi.C
+local base = require "resty.core.base"
+local FFI_ERROR = base.FFI_ERROR
+local tostring = tostring
+
+ffi.cdef[[
+int ngx_http_lua_ffi_get_phase(ngx_http_request_t *r, char **err)
+]]
+
+
+local _M = {
+    _VERSION = base.version
+}
+
+local errmsg = base.get_errmsg_ptr()
+local context_lookup = {}
+context_lookup[0x0001] = "set"
+context_lookup[0x0002] = "rewrite"
+context_lookup[0x0004] = "access"
+context_lookup[0x0008] = "content"
+context_lookup[0x0010] = "log"
+context_lookup[0x0020] = "header_filter"
+context_lookup[0x0040] = "body_filter"
+context_lookup[0x0080] = "timer"
+context_lookup[0x0100] = "init_worker"
+context_lookup[0x0200] = "balancer"
+context_lookup[0x0400] = "ssl_cert"
+context_lookup[0x0800] = "ssl_session_store"
+context_lookup[0x1000] = "ssl_session_fetch"
```

Comment:
silly mistake by me. i didnt realize that brackets could be used to defined keys as integers, in addition to strings. my fault for not understanding the syntax enough here :)

as for re-hashes, i believe this occurs when the table is require'd the first time; i dont know if it happens on subsequent requires (given the context of lua-resty-core, this seems unlikely), so when taking this approach i weighed against `pcall`ing `table.new` (since it would only be called once, a few re-hashes isnt a big deal, and i didnt want to bikeshed).

as for capitalization, im not sure, non-table constants are indeed all-caps, not sure if this applies to a table such as this. ill happily adjust as needed once a definitive, consistent style is declared.